extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -80,6 +80,7 @@ AmpBaseCarousel['children'] = {
   },
   'children': {
     name: 'children',
+    props: {'thumbnailSrc': {attr: 'data-thumbnail-src'}, 'src': {attr: 'src'}},
```

Comment:
This does not do what you would expect (I assumed it would be like the component level props definitions which get processed). The value given to the slot is the literal objects here. We should update with some processing? Or am I using this incorrectly? cc @dvoytenko @jridgewell

Reply:
For now I have moved the `propDef` logic to a new function `parsePropDefs` in `PreactBaseElement` and called it also for the children prop definitions.